src/components/transactions/TxDetails/SafeTxGasForm.tsx

```diff
@@ -20,52 +20,60 @@ const SafeTxGasForm = () => {
     mode: 'onChange',
   })
 
-  const onSubmit = (values: FormFields) => {
+  const onFormSubmit = (values: FormFields) => {
     setSafeTxGas(values.safeTxGas || 0)
-    setEditing(false)
+    onSubmit()
   }
 
   const onBlur = () => {
     setTimeout(() => {
-      setEditing(false)
       formMethods.setValue('safeTxGas', safeTxGas)
+      onSubmit()
     }, 100)
   }
 
   return (
-    <Box display="flex" alignItems="center" gap={1} position="relative">
-      <>
-        {safeTxGas}
+    <Paper sx={{ position: 'absolute', zIndex: 2, p: 1, ml: '-22px' }} elevation={2}>
+      <form onSubmit={formMethods.handleSubmit(onFormSubmit)} style={{ display: 'flex' }}>
+        <NumberField
+          size="small"
+          autoFocus
+          type="number"
+          error={!!formMethods.formState.errors.safeTxGas}
+          sx={{ width: '7em' }}
+          {...formMethods.register('safeTxGas', {
+            valueAsNumber: true,
+            min: 0,
+            setValueAs: Math.round,
+            onBlur,
+          })}
+        />
+        <Button type="submit" size="small" variant="contained" sx={{ ml: 1 }}>
+          Save
+        </Button>
+      </form>
+    </Paper>
+  )
+}
 
-        {isEditable && (
-          <Link component="button" onClick={() => setEditing(true)} fontSize="small">
-            Edit
-          </Link>
-        )}
-      </>
+const SafeTxGasForm = () => {
+  const { safeTx, safeTxGas = 0 } = useContext(SafeTxContext)
+  const { safe } = useSafeInfo()
+  const isOldSafe = safe.version && isLegacyVersion(safe.version)
+  const isEditable = safeTx?.signatures.size === 0 && (safeTxGas > 0 || isOldSafe)
+  const [editing, setEditing] = useState(false)
 
-      {editing && (
-        <Paper sx={{ position: 'absolute', zIndex: 2, p: 1, ml: '-22px' }} elevation={2}>
-          <form onSubmit={formMethods.handleSubmit(onSubmit)} style={{ display: 'flex' }}>
-            <NumberField
-              size="small"
-              autoFocus
-              type="number"
-              error={!!formMethods.formState.errors.safeTxGas}
-              sx={{ width: '7em' }}
-              {...formMethods.register('safeTxGas', {
-                valueAsNumber: true,
-                min: 0,
-                setValueAs: Math.round,
-                onBlur,
-              })}
-            />
-            <Button type="submit" size="small" variant="contained" sx={{ ml: 1 }}>
-              Save
-            </Button>
-          </form>
-        </Paper>
+  return (
+    <Box display="flex" alignItems="center" gap={1} position="relative">
+      {safeTxGas}
+
+      {isEditable && (
+        <Link component="button" onClick={() => setEditing(true)} fontSize="small">
+          Edit
+        </Link>
       )}
+
+      {editing && <Form onSubmit={() => setEditing(false)} />}
```

Comment:
Moving the form into a separate component fixes the bug that the default value wasn't updated when safeTxGas is recalculated.